library/Zend/InputFilter/Input.php

```diff
@@ -326,6 +326,10 @@ public function isValid($context = null)
             return true;
         }
 
+        if ($empty && !$this->isRequired()) {
```

Comment:
I see your point, but the casting should be done in `isRequired()`. Checking for `false` here would mean `null` is considered `true` while `null` is normally considered `false`. That is probably also the reason you get many errors.

Instead of checking everything with `===` or `!==`, it is better to force the methods to return the value which is defined in the docblocks. A method that can return `null` when a the docblock explicitely says `@return bool` is wrong IMO.